### ISSUE TITLE
V24.4 alpha 22 testing: "ConfirmDialog" component should not add unsupported methods

### ISSUE THREAD

**Issue by bennewi** [vaadin/platform#5243]
### Describe the bug

A number of methods were added to the ConfirmDialog component in V24.4 that are marked as "Deprecated, not supported since 24.4" 

This creates an ambiguous and confusing situation.

### Expected-behavior

Better to either add the methods and support them or don't add the methods.

### Reproduction

(seen in api docs)

### System Info

(seen in api docs)

**Comment by bennewi** [comment]
formal java api docs could make that clear. This doesn't have to block beta

**Comment by web-padawan** [comment]
The change is intentional. See https://github.com/vaadin/flow-components/pull/5519 where it was implemented.

**Comment by bennewi** [comment]
I understand, but anyone reading java api docs won't get the explanation. Would be better to add because it looks rather strange at the moment.

**Comment by manolo** [comment]
@web-padawan So what is needed for closing this issue? add some explanation in javadocs?

**Comment by web-padawan** [comment]
Yes, we probably could clarify Javadoc (although it already describes why these methods are unsupported).
